gpytorch/lazy/sum_lazy_tensor.py

```diff
@@ -73,6 +71,22 @@ def __add__(self, other):
             return SumLazyTensor(*(list(self.lazy_tensors) + list(other.lazy_tensors)))
         elif isinstance(other, LazyTensor):
             return SumLazyTensor(*(list(self.lazy_tensors) + [other]))
+        elif isinstance(other, Tensor):
+            # get broadcast shape, assuming mul broadcasting the same as add broadcasting
+            broadcasted_shape = _mul_broadcast_shape(self.shape, other.shape)
+
+            # lazify + broadcast other
+            broadcasted_other = lazify(other.expand(broadcasted_shape))
+
+            # update the lazy tensors' shape as well
+            if broadcasted_shape != self.shape:
+                broadcasted_lts = [
+                    lt.expand(*broadcasted_shape, 1).squeeze(-1).transpose(-1, -2) for lt in self.lazy_tensors
+                ]
+            else:
+                broadcasted_lts = list(self.lazy_tensors)
+
+            return SumLazyTensor(*(broadcasted_lts + [broadcasted_other]))
```

Comment:
Rather than doing the work here, I think it would make sense to add the broadcasting operations in `LazyTensor.__add__` or in the SumLazyTensor constructor. We can also wrap all tensors as LazyTensors as well.

Reply:
A couple of comments here - I'm not sure if it's possible to put the broadcasting in `LazyTensor.__add__` and the SumLazyTensor constructor might require a little bit more work to check all of the lazy tensors.

Is there a reason to not put the broadcasting here?

Reply:
Putting it in `LazyTensor.__add__` or the SLT constructor catches more cases (e.g. if you were to add a ToeplitzLazyTensor and a Tensor), and requires fewer conditional statements which is cleaner/less brittle.

For `LazyTensor.__add__`:

```python
def __add__(self, other):
    other = lazify(other)
    shape = _mul_broadcast_shape(self.shape, other.shape)
    return SumLazyTensor(self.expand(shape), other.expand(shape))
```

For the SLT constructor:

For `LazyTensor.__add__`:

```python
def __init__(self, *lazy_tensors, **kwargs):
    try:
        lazy_tensors = tuple(lazify(lt) for lt in lazy_tensors)
    except TypeError:
        raise TypeError("All arguments of a SumLazyTensor should be LazyTensors or Tensors")
    batch_shape = _mul_broadcast_shape(*[lt.shape for lt in lazy_tensors])
    lazy_tensors = tuple(lt.expand(shape) for lt in lazy_tensors)
    super.__init__(*lazy_tensors, **kwargs)
    self.lazy_tensors = lazy_tensors
```

Reply:
Thanks for the help - I ended up using the sum lazy tensor constructor.